idx/bigtable/bigtable.go

```diff
@@ -474,10 +474,17 @@ func (b *BigtableIdx) Delete(orgId uint32, pattern string) ([]idx.Archive, error
 		}
 	}
 	statDeleteDuration.Value(time.Since(pre))
+
+	// there is nothing higher up in the call path that uses MetricDefinitions
+	// so this is the safest place to release the objects in MetricDefinition
+	// that have been interned
+	for _, arc := range defs {
+		idx.InternReleaseMetricDefinition(arc.MetricDefinition)
+	}
```

Comment:
does this mean the returned defs are now incorrect or can't be read?
if so, let's just return the number of them, as that is the only thing we actually seem to use in call sites, and that way it's clear that these are no longer valid

Reply:
Yeah we can do that. I was attempting to keep the code as intact as possible while writing this PR because I knew it was going to be big, and the fewer changes the easier it would be to review. I agree we should just return the length of defs.